Core/Src/main.c

```diff
@@ -65,6 +68,13 @@ void SystemClock_Config(void);
 int main(void)
 {
   /* USER CODE BEGIN 1 */
+	uint8_t spi_rx_buf[3]; //receive buffer
+	uint8_t spi_tx_buf[3]; //transmit buffer
+	uint16_t adc_val = 0x0000;
+	uint16_t MAX_ADC_VAL = 1023; //equals 2^10 - 1, shouldn't this be higher than MIN_COMPARE_VAL?
+	uint16_t MIN_COMPARE_VAL = 3276; //0.05 * 65535
```

Comment:
Love the naming and use of constants, just add `static const` to the start of the line for these to tell the compiler that these are in fact constant and won't be changed.